### PR TITLE
add possibility to specify crop dimensions

### PR DESCRIPTION
**Pull request by maxfreu**
This adds the possibility to specify crop dimensions. Hope it looks ok :)

### CODE REVIEW

File: src/methods/crop_extend.jl [hunk 1]
```diff
@@ -92,15 +92,18 @@ function _crop_to(x, to; kw...)
         return _crop_to(x, ext; kw...)
     end
 end
-_crop_to(A, to::RasterStackOrArray; kw...) = _crop_to(A, dims(to); kw...)
+function _crop_to(A, to::RasterStackOrArray; kw...)
+    target_dims = get(kw, :dims, name.(dims(to)))
```
Comment (rafaqz) [inline src/methods/crop_extend.jl:96]: You can just do this in the keywords directly without `get`, just add a `dims` keyword.

**Review by rafaqz — CHANGES_REQUESTED** [review]
Lools good

File: src/methods/crop_extend.jl [hunk 2]
```diff
@@ -92,15 +92,18 @@ function _crop_to(x, to; kw...)
         return _crop_to(x, ext; kw...)
     end
 end
-_crop_to(A, to::RasterStackOrArray; kw...) = _crop_to(A, dims(to); kw...)
+function _crop_to(A, to::RasterStackOrArray; kw...)
```
Comment (rafaqz) [inline src/methods/crop_extend.jl:95]:
```suggestion
function _crop_to(A, to::RasterStackOrArray; dims=dims(to), kw...)
```

File: src/methods/crop_extend.jl [hunk 3]
```diff
@@ -92,15 +92,18 @@ function _crop_to(x, to; kw...)
...
+    target_dims = get(kw, :dims, name.(dims(to)))
+    return _crop_to(A, dims(to, target_dims); kw...)
```
Comment (rafaqz) [inline src/methods/crop_extend.jl:97]:
```suggestion
    return _crop_to(A, DD.dims(to, dims); kw...)
```

File: src/methods/crop_extend.jl (hunk 1 above)
Comment (rafaqz) [inline src/methods/crop_extend.jl:96]:
```suggestion
```

**Comment by maxfreu** [comment]
This was a clear case of not seeing the forest for the trees :D I'll fix it!

**Comment by maxfreu** [comment]
I noticed that there are unused variables in the crop functions, so it's likely that they don't work as you intended:

https://github.com/rafaqz/Rasters.jl/blob/88af714ec6194c321c6d1188e2c38c5e318318cf/src/methods/crop_extend.jl#L98-L100

https://github.com/rafaqz/Rasters.jl/blob/88af714ec6194c321c6d1188e2c38c5e318318cf/src/methods/crop_extend.jl#L104

**Comment by rafaqz** [comment]
Good cqtch, the second one `ds` is just no longer needed, thats in DD now.

In the first `sampled` should be used instead of `to`. It will mostly just stop bands or other `Categorical` lookups being used to crop.

**Comment by maxfreu** [comment]
Actually, cropping forward ordered categorical int64 band dimensions worked. But I changed it nevertheless, as there probably will be problems with string values.

**Comment by rafaqz** [comment]
Yeah the idea was you basically never want to use crop for bands or categorical things, so passing in bands in the `to` template should be ignored

**Pull request merged**